include/revng/Support/YAMLTraits.h

```diff
@@ -259,13 +259,23 @@ deserializeImpl(llvm::StringRef YAMLString, void *Context = nullptr) {
 
 } // namespace revng::detail
 
-template<NotTupleTreeCompatible T>
+template<typename RootT>
+struct TupleTreeVisitor;
+
+// The "natural" deserialization of tuple tree roots is forbidden,
+// use `TupleTree<T>::deserialize` instead.
```

Comment:
This sounds weird. Is it forbidden or is there some other specialization that does something more?

Reply:
Add a comment on why it's forbidden (`initializeReferences`).

Reply:
Done!